src/lndk_offers.rs

```diff
@@ -190,6 +196,29 @@ impl OfferHandler {
     }
 }
 
+// pay_invoice tries to pay the provided invoice.
+pub(crate) async fn pay_invoice(
+    mut payer: impl InvoicePayer + std::marker::Send + 'static,
+    path: BlindedPath,
+    cltv_expiry_delta: u16,
+    fee_base_msat: u32,
+    payment_hash: [u8; 32],
+    msats: u64,
+) -> Result<(), OfferError<Secp256k1Error>> {
+    let resp = payer
+        .query_routes(path, cltv_expiry_delta, fee_base_msat, msats)
+        .await
+        .map_err(OfferError::RouteFailure)?;
+
+    // TODO: If route fails, loop over resp.routes array and try alternatives.
```

Comment:
I think that LND has deprecated replying with multiple routes, and this is just an array for backwards compat so this may not apply.

Reply:
Ah yup looks like it's deprecated